src/extensions/default/AutoUpdate/main.js

```diff
@@ -392,50 +426,56 @@ define(function (require, exports, module) {
      * Handles and processes the update info, required for app auto update
      * @private
      * @param {Array} updates - array of {...Update~Entry} update entries
+     * @param {boolean} test - boolean to denote whether the API is invoked for unit testing
      */
-    function _updateProcessHandler(updates) {
+    function _updateProcessHandler(updates, test) {
```

Comment:
Instead of passing a test parameter, we can have explicit test APIs which mock the normal behavior, á la ExtensionLoader.js